app/models/organization_member.rb

```diff
@@ -1,4 +1,28 @@
 class OrganizationMember < ApplicationRecord
   belongs_to :organization
   belongs_to :user
+
+  # while the canonical list of roles is the "organization_member_role" enum in the
+  # database, this enum exists to help display the list of roles
+  # elsewhere in the app
+  enum :role, {
+    admin: "admin",
+    member: "member"
+  }
+
+  validates :full_name, presence: true
+  validates :organization_id, uniqueness: {scope: :user_id}
+
+  def self.create_with_user(email:, **organization_member_attributes)
+    organization_member = new(organization_member_attributes)
+    user = User.find_by_case_insensitive_email(email) || User.new(email:, password: Devise.friendly_token)
```

Comment:
If a user is created here, a confirmation link is automatically sent to them, but they probably need to do a "forgot password" flow to actually confirm themselves and sign in.

Reply:
It might confuse people a bit if they get an email that says they need to confirm but it doesn't work.

OTOH, we're likely going to start testing with single person orgs, so we can improve this invite messaging/flow in the future.